group/curve25519/suite.go

```diff
@@ -49,7 +49,7 @@ func (s *suiteEd25519) NewKey(rand cipher.Stream) crypto.Scalar {
 }
 
 // Ciphersuite based on AES-128, SHA-256, and the Ed25519 curve.
-func NewAES128SHA256Ed25519(fullGroup bool) crypto.Suite {
+func NewAES128SHA256Ed25519(fullGroup bool) *suiteEd25519 {
```

Comment:
Putting this together with Bryan's comment, I'd do:
1. make `suiteEd25519` public and group: `GroupEd25519`
2. add a `SuiteEd25519` with `sha256`

does that make sense?